sale_general_term/models/sale_general_term.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class SaleGeneralTerm(models.Model):
+
+    _name = 'sale.general.term'
+    _description = 'Sale General Term'
+
+    @api.constrains('is_enabled', 'country_ids', 'lang')
+    def _check_is_enabled_country_ids(self):
+        for term in self:
+            enabled_terms = self.search([
+                ('is_enabled', '=', True),
+                ('country_ids', 'in', term.country_ids.ids),
+                ('lang', '=', term.lang)
+            ])
+            if len(enabled_terms) > 1:
+                raise ValidationError(_(
+                    'There are already enabled terms for the Language '
+                    'and at least one of the Countries selected. '
+                    'Each Country/Language is allowed only one '
+                    'enabled General Terms.'
+                ))
+
+    @api.constrains('is_enabled', 'active')
+    def _check_is_enabled_active(self):
+        for term in self:
```

Comment:
I'm not sure this will raise as active is filtering recordsets.

Reply:
The self recordset can have active=False records. I expect to be the case when writing an a record setting the on the active field .
If api.constrains does not work for active=False records, that would be an ORM bug.